### PR TITLE
feat(module:menu): add click trigger for submenus

### PR DESCRIPTION
**Pull request by ParsaArvanehPA**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [✔] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [✔] Tests for the changes have been added (for bug fixes / features)
- [✔] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[✔] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
In Ant design, it is possible to choose between a `click` or a `hover` trigger for submenues.

Issue Number: #4745 

## What is the new behavior?
`click` trigger implemented.

## Does this PR introduce a breaking change?
```
[ ] Yes
[✔] No
```

### CODE REVIEW

File: components/menu/submenu-title.component.ts [hunk 1]
```diff
@@ -96,12 +97,13 @@ export class NzSubMenuTitleComponent implements OnDestroy, OnInit {
   }
 
   setMouseState(state: boolean): void {
-    if (!this.nzDisabled) {
+    if ((!this.nzDisabled && this.nzTriggerSubMenuAction === 'hover') || !state) {
```
Comment (Laffery) [inline components/menu/submenu-title.component.ts:100]: `|| !state`  -> mouseLeave event will trigger closing the submenu

File: components/menu/submenu-title.component.ts [hunk 2]
```diff
@@ -96,12 +97,13 @@ export class NzSubMenuTitleComponent implements OnDestroy, OnInit {
...
       this.subMenuMouseState.next(state);
     }
   }
   clickTitle(): void {
-    if (this.mode === 'inline' && !this.nzDisabled) {
+    if ((this.mode === 'inline' || this.nzTriggerSubMenuAction === 'click') && !this.nzDisabled) {
+      this.subMenuMouseState.next(true);
```
Comment (Laffery) [inline components/menu/submenu-title.component.ts:106]: 1. It'll be great if support closing submenu when clicking outside of the submenu-title
2. Please check that sub-sub-menu will be closed recursively if closing the higher-level submeu. For example, submenu of Option3 should be closed when closing submenu of Option2
```txt
- Option 1
- Option 2
   - Option 3
      - Option 4
```

**Review by Laffery — CHANGES_REQUESTED** [review]
partially approved

File: components/menu/submenu-title.component.ts (hunk 1 above)
Comment (ParsaArvanehPA) [reply components/menu/submenu-title.component.ts]: done

File: components/menu/submenu-title.component.ts (hunk 2 above)
Comment (ParsaArvanehPA) [reply components/menu/submenu-title.component.ts:105]: Good call, both these behavours are now implemented.

**Comment by ParsaArvanehPA** [comment]
> partially approved

your requested changes are made, could you please check it once more

**Comment by ParsaArvanehPA** [comment]
Hey guys
I have some old branches that I want to take care of.
Could you please review this again?
@Laffery 
@HyperLife1119 
@Nicoss54

**Review by Laffery — APPROVED** [review]
LGTM
BTW, I think the API shoud be named `nzSubmenuTrigger`, I will add a commit to change it

**Comment by Laffery** [comment]
> LGTM BTW, I think the API shoud be named `nzSubmenuTrigger`, I will add a commit to change it

Alright, ant-design use `triggerSubMenuAction` too

**Pull request merged**